test/thermo/CoverageDependentSurfPhase_Test.cpp

```diff
@@ -0,0 +1,437 @@
+    double s_int = us.convertFrom(-0.42, "kcal/mol/K");
+    // Parameters for dependency on heat capacity
+    double cp_a = us.convertFrom(0.02e-1, "kJ/mol/K");
+    double cp_b = us.convertFrom(-0.156e-1, "kJ/mol/K");
+    double int_cp_T_tnow = log(test_Ts[3]) * (cp_a * log(test_Ts[3]) + 2 * cp_b);
+    double int_cp_T_298 = log(298.15) * (cp_a * log(298.15) + 2 * cp_b);
+    // Parameter for reference coverage correction
+    double ref_cov = 0.22;
+
+    vector_fp entropies_R(5);
+    vector_fp expected_result(5);
+
+    // Get CoverageDependentSurfPhase standard entropy method value
+    covdepsurf_phase->setTemperature(test_Ts[3]);
+    surfphase_ptr->setCoverages(test_covs[0].data());
+    covdepsurf_phase->getEntropy_R(&entropies_R[0]);
+    // Externally calculate value by adding coverage-dependent terms
+    // to SurfPhase standard entropy method value
+    idealsurf_phase->setTemperature(test_Ts[3]);
+    idealsurf_phase->getEntropy_R(&expected_result[0]);
+    expected_result[1] += (s_slope * test_covs[0][1]) / GasConstant;
+    expected_result[1] += (s_low * 0.4) / GasConstant;
+    expected_result[1] += (s_high * (test_covs[0][2] - 0.4)) / GasConstant;
+    expected_result[1] += poly4(test_covs[0][3], s_coeffs.data()) / GasConstant;
+    expected_result[1] += s_int / GasConstant;
+    expected_result[1] += 0.5 * (int_cp_T_tnow - int_cp_T_298)
+                          * test_covs[0][2] * test_covs[0][2] / GasConstant;
+    expected_result[1] -= -log(ref_cov);
+    EXPECT_NEAR(entropies_R[1], expected_result[1], 1.e-6);
+}
+
+TEST_F(CoverageDependentSurfPhase_Test, standard_cp_R)
+{
+    // This test checks if CoverageDependentSurfPhase standard heat capacity is
+    // properly matches with explicitly calculated standard heat capacity
+
+    // Define coverage-dependent parameters same as given in
+    // copt_covdepsurf_example.yaml for explicit external calculations
+    // Parameters for dependency on heat capacity
+    double cp_a = us.convertFrom(0.02e-1, "kJ/mol/K");
+    double cp_b = us.convertFrom(-0.156e-1, "kJ/mol/K");
+
+    vector_fp cps_R(5);
+    vector_fp expected_result(5);
+
+    // Get CoverageDependentSurfPhase standard heat capacity method value
+    covdepsurf_phase->setTemperature(test_Ts[4]);
+    surfphase_ptr->setCoverages(test_covs[0].data());
+    covdepsurf_phase->getCp_R(&cps_R[0]);
+    // Externally calculating value by adding coverage-dependent terms
+    // to SurfPhase standard heat capacity method value
+    idealsurf_phase->setTemperature(test_Ts[4]);
+    idealsurf_phase->getCp_R(&expected_result[0]);
+    expected_result[1] += (cp_a * log(test_Ts[4]) + cp_b)
+                           * test_covs[0][2] * test_covs[0][2] / GasConstant;
+    EXPECT_NEAR(cps_R[1], expected_result[1], 1.e-6);
+}
+
+TEST_F(CoverageDependentSurfPhase_Test, standard_gibbs_RT)
+{
+    // This test checks if standard gibbs free energy is
+    // properly matches value calculated with enthalpy and entropy
```

Comment:
This test and I think all of the ones after it are unnecessary, and fully covered by the tests executed by `consistency.cpp`.